src/main/perl/Path.pm

```diff
@@ -214,17 +216,20 @@ sub _safe_eval
     %opts = %$optsref if $optsref;
 
     local $@;
-    eval {
-        $res = $funcref->(@args, %opts);
-    };
+    my $status = eval {
+                       $res = $funcref->(@args, %opts);
+                      };
 
-    if ($@) {
-        chomp($@);
-        return $self->fail("$failmsg: $@");
-    } else {
-        my $res_txt = defined($res) ? "$res" : '<undef>';
-        chomp($res);
+    # $status is undef if there is a syntax or runtime error or if $res is undef.
+    if ( defined($status) ) {
```

Comment:
I understand now my mistakes and why I spent so much time to cope with the exceptions returned! Fixed!